src/components/BestProduct.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from "react";
+import { fetchApi } from "../api/axiosInstance";
+import likeButton from "../images/etc/likeButton.svg";
+import nonImage from "../images/etc/nonImage.svg";
+import "../css/BestProduct.css";
+
+const PCBEST = 4;
+const TABLETBEST = 2;
+const MOBILEBEST = 1;
+
+const getPageSize = () => {
+  const width = window.innerWidth;
+  // 전체 상품의 반응형 그리드 설정
+  if (width >= 1200) {
+    return PCBEST;
+  } else if (width >= 744) {
+    return TABLETBEST;
+  } else {
+    return MOBILEBEST;
+  }
+};
```

Comment:
@gahyeon-yang  요기 width를 인자로 받도록 하면 어떨까요?

```suggestion
const getPageSize = (width) => {
  // 전체 상품의 반응형 그리드 설정
  if (width >= 1200) {
    return PCBEST;
  } else if (width >= 744) {
    return TABLETBEST;
  } else {
    return MOBILEBEST;
  }
};
```
```js
getPageSize(window.innerWidth)
```